src/cpp/editor/editor.cpp

```diff
@@ -63,4 +64,59 @@ void editManager::InitializeMap(int random, int width, int height) {
 		undoStack.pop();
 	}
 	this->InitializeMap_orig(random, width, height);
+}
+
+void editManager::SpellScrollEditDialog(int *RelatedToSpellIdx) {
+  int i; // [sp+38h] [bp-4h]@4
+  const int WINDOW_LABEL = 101;
+  const int DROPLIST_LABEL = 102;
+  const int FIELD_ID_SPELL_SCROLL_PAYLOAD = 100;
+  const int KEYCODE_UNKNOWN_1 = 54;
+  OriginalSpell = *RelatedToSpellIdx;
+  gpCellEditDialog = new heroWindow(0, 0, "x_spedit.bin");
+  
+  GUISetText(gpCellEditDialog, WINDOW_LABEL, "Spell Scroll");
+  GUISetText(gpCellEditDialog, DROPLIST_LABEL, "Attach Spell");
+  for (i = 0; i < NUM_SPELLS; ++i) {
+    GUIDroplistAdd(gpCellEditDialog, FIELD_ID_SPELL_SCROLL_PAYLOAD, gSpellNames[i]);
+  }
+  GUIBroadcastMessage(gpCellEditDialog, FIELD_ID_SPELL_SCROLL_PAYLOAD, KEYCODE_UNKNOWN_1, (void *)*RelatedToSpellIdx);
+  gpWindowManager->DoDialog(gpCellEditDialog, SpellScrollEditDialogCallback, 0);
+  delete(gpCellEditDialog);
+  if (gpWindowManager->buttonPressedCode != BUTTON_CANCEL) {
+    *RelatedToSpellIdx = OriginalSpell;
+    gpEditManager->setOnEventUpdate = 1;
+  }
+  gpEditManager->UpdateCursor();
+  RedrawEditPane();
+}
+
+int __fastcall SpellScrollEditDialogCallback(tag_message& msg) {
+  const int FIELD_ID_SPELL_SCROLL_PAYLOAD = 100;
+  const int KEYCODE_UNKNOWN_2 = 10;
+
+  if (msg.eventCode == INPUT_KEYDOWN_EVENT_CODE) {
+    if (msg.xCoordOrKeycode == 1) {
+      msg.eventCode = INPUT_GUI_MESSAGE_CODE;
+      msg.yCoordOrFieldID = KEYCODE_UNKNOWN_2;
+      msg.xCoordOrKeycode = msg.yCoordOrFieldID;
+      return 2;
```

Comment:
I found GUI_MESSAGE_REPAINT and it makes sense to me. I tried to look at the decompiled code, but according to that, the game should get stuck in an endless loop if "2" isn't returned. That seems off to me. There are also some other odd logic things in the function that I traced back to, "DoDialog" (of heroWindowManager, not executive), so I think that the decompilation could be off. Either way, this constant makes the most sense to me out of all of them. It actually is the only one that makes any sense.